it/src/main/java/com/google/cloud/teleport/it/spanner/DefaultSpannerResourceManager.java

```diff
@@ -0,0 +1,244 @@
+    return new Builder(testId, projectId, region);
+  }
+
+  private synchronized void createInstance() {
+    LOG.info("Creating instance {} in project {}.", instanceId, projectId);
+    InstanceInfo instanceInfo =
+        InstanceInfo.newBuilder(InstanceId.of(projectId, instanceId))
+            .setInstanceConfigId(InstanceConfigId.of(projectId, region))
+            .setDisplayName(instanceId)
+            .setNodeCount(1)
+            .build();
+    try {
+      instanceAdminClient.createInstance(instanceInfo).get();
+      hasInstance = true;
+      LOG.info("Successfully created instance {}.", instanceId);
+    } catch (ExecutionException | InterruptedException e) {
+      throw new SpannerResourceManagerException("Failed to create instance.", e);
+    }
+  }
+
+  private synchronized void createDatabase() {
+    LOG.info("Creating database {} in instance {}.", databaseId, instanceId);
+    try {
+      databaseAdminClient.createDatabase(instanceId, databaseId, Collections.emptyList()).get();
+      databaseClient = spanner.getDatabaseClient(DatabaseId.of(projectId, instanceId, databaseId));
+      LOG.info("Successfully created database {}.", databaseId);
+    } catch (ExecutionException | InterruptedException e) {
+      throw new SpannerResourceManagerException("Failed to create database.", e);
+    }
+  }
+
+  private void checkIsUsable() throws IllegalStateException {
+    if (spanner.isClosed()) {
+      throw new IllegalStateException("Manager has cleaned up all resources and is unusable.");
+    }
+  }
+
+  private void checkHasDatabase() throws IllegalStateException {
+    if (databaseClient == null) {
+      throw new SpannerResourceManagerException("Database does not exist");
+    }
+  }
+
+  @Override
+  public synchronized void createTable(String statement) throws IllegalStateException {
+    checkIsUsable();
+    if (!hasInstance) {
+      createInstance();
+    }
+    if (databaseClient == null) {
+      createDatabase();
+    }
+    LOG.info("Creating table in database {} using statement '{}'", databaseId, statement);
+    try {
+      databaseAdminClient
+          .updateDatabaseDdl(
+              instanceId, databaseId, ImmutableList.of(statement), /* operationId= */ null)
+          .get();
+      LOG.info("Successfully created table in database {}.", databaseId);
+    } catch (ExecutionException | InterruptedException e) {
+      throw new SpannerResourceManagerException("Failed to create table.", e);
+    }
+  }
+
+  @Override
+  public synchronized void insertTableRecord(Mutation tableRecord) throws IllegalStateException {
+    insertTableRecords(ImmutableList.of(tableRecord));
+  }
+
+  @Override
+  public synchronized void insertTableRecords(Iterable<Mutation> tableRecords)
+      throws IllegalStateException {
+    String tableId = Iterables.get(tableRecords, 0).getTable();
+    LOG.info(
+        "Inserting {} records into table {}.{}.{}",
```

Comment:
There's actually nothing about this implementation requires users to insert into a single table, and the interface's doc comment states that updating an existing record is possible.

I think we should just rename this to `write` and let users do whatever they intend to do. If we ever want more granular control, we can add that, but it's harder on our end and may not offer much benefit to usability. The log statement would update to something like: "Sending {} mutations to {}.{}" Basically, leave off the table and just make a generic reference to the number of mutations, not a specific type of mutation.